src/shared/awsClientBuilder.ts

```diff
@@ -137,3 +138,29 @@ export class DefaultAWSClientBuilder implements AWSClientBuilder {
         return service
     }
 }
+
+type Token = CancellationToken | ((request: AWS.Request<any, AWSError> & RequestExtras) => CancellationToken)
+export function addCancellationToken(input: Token, logger?: (msg: string) => void): RequestListener {
```

Comment:
This allows SDK requests to be cancelled.
It wouldn't be sustainable though to implement it for everything like how I did for `getFederationToken`. We'd need to start modifying the client prototype for something that can scale.